src/platform/tls_openssl.c

```diff
@@ -1129,7 +1148,8 @@ CxPlatTlsSecConfigCreate(
     }
 
     if (CredConfigFlags & QUIC_CREDENTIAL_FLAG_CLIENT) {
-        SSL_CTX_set_verify(SecurityConfig->SSLCtx, SSL_VERIFY_PEER, CxPlatTlsCertificateVerifyCallback);
+        SSL_CTX_set_cert_verify_callback(SecurityConfig->SSLCtx, CxPlatTlsCertificateVerifyCallback, NULL);
```

Comment:
So this will only call the verify on the leaf certificate, and not for each certificate in the chain?

Reply:
Correct. However inside of verify it will grab the entire chain and verify it. This was being done before as well, as we would always grab the leaf certificate at the beginning of every callback. So we were just repeating the same work for every certificate in the chain.